tests/conftest.py

```diff
@@ -47,9 +47,26 @@ def restore_env_variables():
     """Ensures that environment variables set during the test do not leak out."""
     env_backup = os.environ.copy()
     yield
+    leaked_vars = os.environ.keys() - env_backup.keys()
     # restore environment as it was before running the test
     os.environ.clear()
     os.environ.update(env_backup)
+    # these are currently known leakers - ideally these would not be allowed
+    allowlist = {
+        "CUDA_DEVICE_ORDER",
+        "LOCAL_RANK",
+        "NODE_RANK",
+        "WORLD_SIZE",
+        "MASTER_ADDR",
+        "MASTER_PORT",
+        "PL_GLOBAL_SEED",
+        "PL_SEED_WORKERS",
+        "WANDB_MODE",
+        "HOROVOD_FUSION_THRESHOLD",
+        "RANK",  # set by DeepSpeed
+    }
```

Comment:
```suggestion
        "RANK",  # set by DeepSpeed
        "POPLAR_ENGINE_OPTIONS",  # set by IPUPlugin
    }
```
could also remove it on teardown?